pkg/controller/jobframework/reconciler.go

```diff
@@ -866,9 +878,14 @@ func (r *genericReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 }
 
 func (r *genericReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	b := ctrl.NewControllerManagedBy(mgr).
-		For(r.newJob().Object()).
-		Owns(&kueue.Workload{}, builder.MatchEveryOwner)
+	b := ctrl.NewControllerManagedBy(mgr)
+	if r.newJobEventHandler == nil {
+		b = b.For(r.newJob().Object()).Owns(&kueue.Workload{})
+	} else {
+		eventHandler, name := r.newJobEventHandler(mgr.GetClient())
+		b = b.Watches(r.newJob().Object(), eventHandler).
+			Named(name)
```

Comment:
is the name necessary?

Reply:
Yes, if `For` is called, controller name is composed from the object's GVK. If there's no for, one needs to provide a name:
https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/builder/controller.go#L337